index.js

```diff
@@ -480,16 +483,51 @@ export default function(olLayer, glStyle, source, resolutions, spriteData, sprit
                   style = styles[stylesLength] = new Style();
                 }
                 style.setGeometry(styleGeom);
-                iconImg = iconImageCache[icon];
+                var iconSize = paint['icon-size'](zoom, properties);
+                var iconColor = paint['icon-color'](zoom, properties);
+                var icon_cache_key = icon + '.' + iconSize + '.' + iconColor;
+                iconImg = iconImageCache[icon_cache_key];
```

Comment:
Is `icon-size` supported or used at all for non-circle icons?

Reply:
yeah it is see: https://www.mapbox.com/mapbox-gl-js/style-spec/#layout-symbol-icon-size

it works, although ofcourse it is not very sharp